<a href="https://colab.research.google.com/github/angyf/Entrega_Final_Procesamiento_DanielAlvarez_AngelicaFuentes_IvonneRamirez/blob/main/JupyterNotebook.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#INTRODUCCIÓN 
<div align="justify">

Las ECV son las primer cuasa de muerte en la población adulta mexicana. Se ha determinado que son varios los factores biológicos y conductuales que propician el dasarrollo de ECV, como las enfermedades coronarias entre las más comunes. Dentro de estos factores se encuentran el sobepeso, la obesidad, causado por una mala alimentación y falta de actividad física, padecer diabetes, hipertensión, altos niveles de glucosa, colesterol, hiperlipidemias, el tabaquismo, alcoholismo. 

A pesar de los planes de acción establecidos por el gobierno como, programas te atención a la salud, promoción de actividad física, regímnes alimenticios estrictos, aumento en impuestos de comesibles procesados y cigarrillos entre otros, el número de prevalencia y mortalidad de estas afecciones sigue en aumento. 
Y es que una de las principales causas de la alta mortalidad es el diagnóstico tardío de enfermedades cardiacas, es decir, cuando una persona es diagnosticada con enfermedad coronaria, su pronóstico de vida o tiempo de supervivencia es muy bajo. 
Por lo anterior surge la importancia de determinar un pronóstico temprano de las ECV.

Es por ello que con este análisis nos enfocaremos en determinar qué factores podrían estar implicados de manera importante en el desarrollo de la enfermedad coronaria (una de las ECV más comunes), y usarlos para predecir qué pacienes son más propensos a tener el riesgo de enfermedad, lo que ayudaría a disminuir los índices de mortalidad.

#ANTECEDENTES

#<center>EPIDEMIOLOGÍA EN MÉXICO: INCIDENCIA, PREVALENCIA Y MORTALIDAD</center>
<div align="justify">

En México, así como en el mundo, las defunciones causadas por enfermedades cardiovasculares (ECV) han ocupado el primer lugar entre las principales causas durante varios años, entre ellas, destacan las enfermedades isquémicas del corazón que presentan una alta incidencia entre la población que fallece a partir de los 45 años **[1]**.

Los factores de riesgo que se asocian con las ECV son la
edad, sexo, tabaquismo, diabetes, hipertensión arterial, obesidad, sobrepeso y sedentarismo, pero la lesión anatomopatológica fundamental es la aterosclerosis (acumulación de grasas, colesterol y otras sustancias dentro y sobre las paredes de las arterias, lo cual puede restringir el flujo sanguíneo), que se presenta con el transcurso de los años y provoca repentinamente daños coronarios y cerebrovasculares. Estos factores de riesgo se clasifican en modificables y no modificables según la posibilidad de influir sobre ellos **[2]**. 

Aunado a esto, en México se ha logrado una disminución de las enfermedades infecciosas, a través de mejorar la salud materno infantil entre otras acciones y esto se ve reflejado en un aumento de la esperanza de vida, la cual se sitúa actualmente en los 75 años, esto se traduce en un envejecimiento de la población y por ende a una mayor exposición de los factores de riesgo anteriormente mencionados por lo que se relaciona indirectamente con el número de casos con ECV **[3]**. Se ha visto que más del 60 % de la población adulta en México presenta al menos algún factor de riesgo de ECV, que junto al proceso de envejecimiento de la población genera un incremento de enfermedades asociadas a la edad, como son las ECV **[4]**.

Asimismo, se han observado incrementos en la prevalencia de obesidad, hipercolesterolemia, hipertensión arterial y síndrome metabólico, que de acuerdo con las encuestas nacionales destacan el sobrepeso y la obesidad, cuya prevalencia en México se ha incrementado con el tiempo hasta convertirse actualmente en la más alta del mundo, constituyendo el principal factor de riesgo cardiovascular modificable **[4]**. 
De acuerdo a la Encuesta Nacional de Salud y Nutrición (ENSANUT), en 2018 el porcentaje de adultos de 20 años o más, con sobrepeso y obesidad, es de 75.2% (39.1% y 36.1% respectivamente), porcentaje que en 2012 fue del 71.3 %. En tanto que la hipertensión pasó de un 16.6% en 2012 a 18.4% en 2018 **[5]**. 

En México, el Sistema Nacional de Salud afirma que la cardiopatía isquémica y la diabetes mellitus, enfermedades relaconadas a ECV, son dos de los más grandes problemas de la salud pública. La cardiopatía afecta sobre todo a los hombres de mediana y avanzada edad y su mortalidad es 20% más alta en los hombres que en las mujeres. El mayor número de defunciones ocurre en personas de ambos géneros de 65 años o más, lo que subraya que la edad avanzada es un poderoso inductor de esta enfermedad. 
La pirámide poblacional determina que la mayoría de los adultos (75%) tiene menos de 55 años y a pesar de que la prevalencia de los factores de riesgo cardiovascular es mayor después de los 40 años, la población que padece alguno de los factores de riesgo, se ubica en la población económicamente activa, con sus consecuencias negativas, tanto a nivel social, a nivel económico y en la calidad de vida. Así, las ECV caen dentro del rubro de gastos excesivos **[6]**. 

Algunos datos generales:
	
*   Las enfermedades no transmisibles (ENT), incluyendo enfermedades cardiovasculares, se estima que representan el 77% del total de muertes de adultos en México.  

*   Las ECV representan casi una cuarta parte (24%) de estos decesos.  

Algunos de los factores de riesgo, relacionados con las enfermedades cardiovasculares en adultos en México, se describen a continuación:  

*   17% de la población es fumadora.
*   22.8% tienen hipertensión, que puede aumentar el riesgo de ataque O insuficiencia cardíaca, enfermedad renal o derrame cerebral. 
*   1 de cada 3 adultos (32.1%) adultos en México son obesos.
*   En el año 2014 casi un tercio de las mujeres (32.7%) fueron clasificados como personas obesas; la obesidad es el factor de riesgo cardiovascular más común entre las mujeres **[7]**.

Como marco de referencia para comprender el impacto que tienen estas enfermedades, se puede comparar con la situación de pandemia que se vive en la actualidad. De acuerdo a datos del INEGI, las defunciones por COVID-19, en el periodo de enero a agosto de 2020, ocupan la segunda causa de muerte a nivel global con 108 658 casos, por debajo de las enfermedades del corazón que ocupan el primer lugar con 141 873 y por encima de la diabetes mellitus que ocupa el tercer lugar con 99 733.

Del número de muertes causadas por enfermedades del corazón, la mayoría tuvieron lugar en personas del sexo masculino con 78 929 muertes, es decir, 16 216 más que el sexo femenino que registró 62 713 defunciones. Cabe mencionar que, si bien el número de muertes por ECV en hombres es mayor, en las mujeres representan 22.32% del total de defunciones, mientras en los hombres representan 19.64%. 
Aunado a esto, se había realizado una proyección de un acumulado de 100 624 muertes por ECV y ocurrieron 139 436 defunciones, por lo que el exceso de mortalidad en este rubro es de 38 812 decesos, lo que representa un 38.6% más de lo proyectado **[1]**.

A continuación, se describen algunos de los principales factores de riesgo mencionados, factores biológicos y conductuales, cómo ha afectado a la población, y qué se ha hecho para atacar el problema desde diferentes planes de acción.



###**<center>1. FACTORES BIOLÓGICOS: SOBREPESO Y OBESIDAD**</center>


Las ECV (enfermedades cardiovasculares) son la mayor causa de muerte en el mundo, han ocasionado cerca de un tercio del total de muertes a nivel mundial **[11]**, y su asociación con la obesidad pandémica y la creciente conducta sedentaria, lo vuelve un problema de interés global tanto de salud como económico, siendo que la obesidad también representa altos costos  médicos. Desde el año de 1980, la prevalencia de obesidad se ha duplicado en alrededor de 70 países incluyendo México **[12]**. 
En nuestro país, el sobrepeso y la obesidad han aumentado en todas las edades, regiones y grupos socioeconómicos, lo que nos ha llevado a ocupar el segundo lugar en el mundo en obesidad en adultos y el primer lugar en obesidad infantil. En 2016, el 39% de las personas adultas de 18 o más años tenían sobrepeso, y el 13% eran obesas. Lo anterior implica también un problema económico para nuestro país puesto que se consume un porcentaje muy elevado del presupuesto en salud. Estimados  en  151  894  millones  de  pesos  sólo  en  2014,  lo  cual  equivale  a  34%  del  gasto  público  en  salud  y  causa  una  pérdida  de  productividad  estimada  en 71 669 millones de pesos (0.4% del pib) por año **[13]**.
El sobrepeso y la obesidad se definen como una acumulación anormal o excesiva de grasa que puede ser perjudicial para la salud. El índice de masa corporal (IMC) es un indicador de la relación entre el peso y la talla que se utiliza frecuentemente para identificar el sobrepeso y la obesidad en los adultos. Se calcula dividiendo el peso de una persona en kilos por el cuadrado de su talla en metros (kg/m2) **[14]**.
En el caso de los adultos, la OMS define el sobrepeso y la obesidad como:
	•	sobrepeso: IMC igual o superior a 25.
	•	obesidad: IMC igual o superior a 30.
El IMC proporciona la medida más útil del sobrepeso y la obesidad en la población, pues es la misma para ambos sexos y para los adultos de todas las edades. Sin embargo, hay que considerarla como un valor aproximado porque puede no corresponderse con el mismo nivel de grosor en diferentes personas.
La causa fundamental del sobrepeso y la obesidad es un desequilibrio energético entre calorías consumidas y gastadas, es decir, un aumento en la ingesta de alimentos de alto contenido calórico que son ricos en grasa y un descenso en la actividad física debido a la naturaleza cada vez más sedentaria de muchas formas de trabajo, los nuevos modos de transporte y la creciente urbanización **[14]**.
Cabe señalar que algunos individuos con sobrepeso que tienen un alto IMC, no tienen evidencia de padecer de factores de riesgo asociados a ECV, a estas personas se dice que tienen una obesidad metabólica saludable (OMS) **[15-16]**. 
Sin embargo, un estudio hecho con una cohorte de 3.5 millones de individuos encontrarorn que, después de un seguimiento medio de 5.4 años, las personas con OMS tenían un riesgo mayor de padecer una enfermedad arterial coronaria, enfermedad cerebrovascular o falla cardiaca con respecto a personas metabólicamente sanas no obesas **[17]**. 
Lo anterior sugiere que la obesidad, es un factor que requiere evaluación médica y tratamiento adecuado, puesto que también se sabe que las personas con OMS tienen un mayor riesgo del 24-33% de mortalidad por ECV comparadas con personas metabólicamente sanas no obesas **[18]**. También se ha encontrado que personas con OMS que mantienen o incrementan su peso, pueden presentar mayores niveles de glucosa, triglicéridos, mayor presión sanguínea, y tener menores niveles de colesterol HDL en comparación con personas metabólicamente sanas no obesas **[19]**. Todos estos factores anteriormente mencionados, están directamente relacionados con el desarrollo de algunas comorbilidades como las que se mencionan a continuación.

  
	•	Hipertensión.

Estudios realizados en diferentes poblaciones del mundo, han mostrado una relación casi lineal entre el IMC y la presión sistólica sanguínea. No obstante, pareciera haber una diferencia respecto al sexo, puesto que mujeres en etapa pre-menopausica tienen menor riesgo de desarrollar ECV en comparación con hombres en el mismo grupo de edad, lo que ha sugerido que los estrógenos tienen un papel cardio-protector **[20]**. Aún así, para ambos sexos, y de acuerdo con el Colegio Americano de Cardiología y los lineamientos para el manejo de la presión sanguínea, la pérdida de peso es crucial y altamente recomendado, el cual debe realizarse mediante la combinación de reducir la ingesta calórica e incrementar de manera sustacial la actividad física, con ello, el mejoramiento de la presión sanguínea es consistente **[21]**. 

	•	Enfermedad arterial coronaria (EAC)

Se ha encontrado que la obesidad está asociada con la EAC independientemente a otros factores de riesgo, lo que se ha sustentado con estudios realizados en hombres jóvenes ya fallecidos (por causas externas) de edad entre 15 a 34 años, donde se ha reportado que un alto IMC se asocia con un alto número de lesiones en la arteria coronaria y con un grado microscópico de aterosclerosis en la misma arteria, con lo que se concluye que la obesidad en la adolescencia y en la juventud, no solo se asocia con el desarrollo de aterosclerosis de manera temprana, sino, que puede predecirse tambien la morbilidad y mortalidad por EAC **[22]**. Es importante mencionar que la obesidad también se asocial con múltiples factores de riesgo para el desarrollo de EAC, como hipertensión, dislipidemias, síndrome metabólico y diabetes mellitus. Las dislipidemias se han considerado como el factor de riesgo más importante para sufrir infartos al miocardio. Diversos estudios han mostrado evidencia de que la pérdida de pes puede revertir los factores de riesg asociados a EAC y potencialmente reducir la mortalidad **[23-24]**. 

	•	Falla cardiaca
La obesidad es el principal factor de riesgo para el desarrollo de una falla cardiaca. Estudios previos han mostrado que el IMC se correlaciona con el riesgo de falla cardiaca; el riesgo incrementa en un 5% para los hombres y en un 7% para las mujeres por cada unidad que aumente el IMC. Esta correlacion positiva se ha confirmado con estudios donde participantes con sobrepeso y obesidad tienen un 49% y 180% de mayor riesgo de falla cardiaca respectivamente con respecto a pacientes en su peso ideal **[25]**. 

	•	Actividad física para la pérdida de peso
Aunque un tiempo mínimo de ejercicio físico de 150 min/semana (moderado) o 75 min/semana (vigoroso) puede mejorar la salud cardiovascular, estos tiempos son inadecuados para una pérdida de peso clínicamente significativa **[26]**. Ya que a pesar de que programas consistentes en recomendaciones de salud publica, típicamente producen la pérdida resultante de (~ 2 kg), una pérdida significativa debe ser de ≥ 5% del peso total con lo que se reduciría de manera efectiva el riesgo de desarrollar ECV **[27]**.


###**<center>2. FACTORES CONDUCTUALES: TABAQUISMO, ALCOHOLISMO Y RÉGIMEN ALIMENTICIO**</center>
	

De acuerdo con la organización mundial de la salud entre los factores de riesgo más importantes de enfermedades cardiovasculares se encuentran el consumo nocivo de tabaco y alcohol, así como una dieta malsana **[28]**.

	•	TABAQUISMO
Ha sido reportado que el tabaquismo aumenta el riesgo de padecer ECV más allá que otros factores de riesgo **[29]** pues se estima que la exposición al humo de tabaco, así como su consumo, son responsables del 17% de las muertes por enfermedades del corazón a nivel mundial, además es la segunda causa de enfermedades cardiovasculares después de la hipertensión arterial. 
En nuestro país durante el año 2017 el 13% de las muertes por enfermedades isquémicas al corazón fueron causadas por tabaquismo **[30]**. Así mismo, de acuerdo con la Encuesta nacional de salud y nutrición, la prevalencia de consumo actual de tabaco entre adultos (personas mayores a 19 años) es de 17.9% en donde la razón hombre:mujer es de 3, esto quiere decir que la población masculina fumadora es tres veces mayor a la femenina **[31]**.
A pesar de que el consumo nocivo de tabaco aumenta el riesgo de la presencia de ECV, hasta ahora no se ha logrado establecer que una reducción del consumo de cigarrillos reduzca notablemente el riesgo de sufrir una enfermedad cardiovascular, sin embargo, la duración de la presencia del tabaquismo si presenta mayor riesgo por lo que el dejar de fumar reduce el riesgo de morbilidad y mortalidad cardiovascular en personas que presenten o no cardiopatía coronaria **[32]**.
De manera general la OMS ha creado el convenio marco de la OMS para el control del tabaco, que contiene directrices y requisitos para la implementación de las medidas de control del tabaco. Algunas disposiciones clave en este convenio son: elevar impuestos sobre productos de tabaco, protección contra la exposición al humo de tabaco ajeno con la creación de espacios libres de tabaco, advertencias en el empaquetado de cigarrillos y prohibición de publicidad de estos productos. Así mismo este tratado contempla otras cuestiones entre ellas la venta a menores y por menores de estos productos **[33]**.

	•	ALCOHOLISMO

Mundialmente de todas las muertes atribuibles al alcohol el 19% fueron asociadas a enfermedades cardiovasculares **[34]**. Aunque el consumo en dosis pequeñas de alcohol reduce el riesgo de tener problemas cardiacos, se sigue debatiendo cual ha de ser el tamaño de dicha dosis para que resulte beneficiosa ya que se ha comparado que el riesgo de una enfermedad coronaria aumenta si el consumo es mayor a 20g diarios de alcohol, pues el beber grandes cantidades en una sola ocasión aumenta el riesgo de arritmias cardiacas y muerte repentina por problemas coronarios, así como el riesgo de hipertensión **[35]**.
En la población adulta mexicana existe una prevalencia de 94.3% donde no se observa diferencia entre el número de hombres y mujeres consumidores. Sin embargo, el 35.5% de la población se refiere a personas con consumo excesivo (cinco o más copas por ocasión en el último mes en el caso de los hombres y cuatro o más en el caso de las mujeres) **[31]**. La diferencia entre el número de copas ingeridas varia por sexo ya que las mujeres parecen estar más expuestas a sufrir daños crónicos producidos por el alcohol que se ha asociado con el hecho de que tienen una cantidad inferior de agua corporal en relación con su peso. 
Al igual que el tabaquismo, el riesgo de sufrir una ECV reduce si la persona suspende la ingesta de alcohol, sin embargo, no elimina completamente el riesgo, por lo que un consumo prolongado de alcohol presenta mayor riesgo **[34]**.

	•	RÉGIMEN ALIMENTICIO

La ingesta adecuada de nutrientes ayuda a prevenir la presencia de enfermedades cardiovasculares y algunos otros factores como la obesidad, sin embargo, la mejora de los hábitos dietéticos es un problema de la sociedad y no solo de cada uno de los individuos. Dentro de las recomendaciones generales para lograr una dieta saludable incluyen aumentar el consumo de frutas y verduras, así como reducir la ingesta calórica de grasas, azucares y el consumo de sal **[36]**.
En México 55.5% de los hogares no cuentan con seguridad alimenticia y el consumo de alimentos y bebidas azucaradas es muy frecuente tanto en niños como en adultos. Así mismo la venta de productos industrializados es cada vez mayor pues a pesar de contar con regulaciones sobre el etiquetado de productos procesados, esto no resulta útil cuando más del 75% de la población no conoce cuál debe ser su consumo promedio diario de calorías o cuando el 10.2% de la población no sabe leer, es por ello que esto implica un problema social **[31]**.
Dentro de la dieta alimentaria uno de los factores que se remarca es ingesta de sodio ya que se estima que mundialmente podría evitarse la defunción de 2.5 millones de personas anualmente si el consumo de sal redujera, esto debido a que un consumo elevado de sodio contribuye a la hipertensión arterial y aumenta el riesgo de cardiopatías y accidentes cerebrovasculares, por lo que un consumo menor a 5 gramos diarios en adultos contribuya a disminuir estos riesgos por lo que se necesita crear entornos donde la población tenga acceso a alimentos salubres y nutritivos **[37]**.
Con el fin de brindar a la población mayor conocimiento sobre buenos hábitos alimenticios, la Organización Panamericana de la Salud brinda liderazgo y cooperación técnica a los Estados Miembros en materia de nutrición saludable, así como también se encarga de promover, generar y difundir información sobre estos temas clave, políticas e intervenciones **[38]**. En nuestro país la medida más es el  etiquetado frontal de advertencia para alimentos y bebidas que ofrecerá a los consumidores información clara y sencilla sobre el exceso de nutrientes en los alimentos procesados **[39]**.


#<center>PLANTEAMIENTO DEL PROBLEMA


La Organización Mundial de la Salud ha estimado que ocurren 12 millones de muertes en todo el mundo, cada año debido a enfermedades cardíacas y en México, las ECV ocupan el primer lugar entre las principales causas durante por enfermedades no transmisibles. A pesar de los planes de acción establecidos por el gobierno como, programas te atención a la salud, promoción de actividad física, regímnes alimenticios estrictos, aumento en impuestos de comesibles procesados y cigarrillos entre otros, el número de prevalencia y mortalidad de estas afecciones sigue en aumento. 
Dado este contexto, el análisis de la mortalidad por ECV en México se vuelve prioritario, para lo que se debe medir su impacto en la salud de la población. Estudiar su tendencia en el tiempo permite comprender cómo el estado de salud de la población cambia y posibilita a las instituciones replantear programas preventivos para intentar mitigar su efecto negativo. Otra forma de ayudar en la resolución del problema de la alta mortalidad, es determinar un pronóstico temprano de las ECV, ayudando a tomar decisiones sobre cambios en el estilo de vida de pacientes de alto riesgo y, a su vez, reducir las complicaciones de forma más personalizada y directa.
Es por ello que en este análisis nos enfocaremos en determinar qué factores podrían estar implicados de manera importante en el desarrollo de la enfermedad coronaria crónica (una de las ECV más comunes), y usarlos para predecir qué pacienes son más propensos a tener el riesgo de enfermedad.

#<center>PREGUNTAS DE INVESTIGACIÓN

Como parte de las preguntas que nos interesa responder acerca de nuestra población de estudio son:

* ¿Qué porcentaje de la poblacion tiene algún factor de riesgo de los ya mencionados?
* ¿Qué porcentaje de hombres y mujeres conforman la población con ECV?
* ¿Cuántos hombres y cuántas mujeres ya sufren de una ECV?
* ¿Cuál es la edad promedio de nuestra población de estudio? ¿Se encuentran dentro de la poblacón de riesgo ya reportada?
* ¿Cómo son los niveles promedio de colesterol, glucosa, presión sangínea, IMC de la población? ¿En su mayoría están fuera de lo normal?
* ¿Qué porcentaje de la población es fumadora? ¿Quiénes fuman más, los hombres o las mujeres?
* ¿Cuántas personas son alcohólicas?
* ¿El alcoholismo podría ser una variable predictiva para el desarrollo de EVC?
* ¿Qué variables podrían estar más asociadas al riesgo de desarrollo de ECV?
* ¿Existen difrencias significativas entre poblaciones respecto a niveles de los diferentes factores de riesgo?
* ¿Se podría construir un modelo que ayude a la predicción de desarrollar una ECV?



**De acuerdo a los análisis realizados previamente en otro grupo poblacional de prueba, encontramos hallazgos que nos permitieron plantear la siguiente hipótesis.**

#<center>HIPÓTESIS
Los principales factores de riesgo que se asocian significativamente al desarrollo de ECV y que ayudan a la predicción de padecer o no una enfermedad cardiaca son: el sexo, la edad, fallas cardiacas, ser fumador, tener altos nivel de colesterol, alto IMC y alta presión sanguínea.



#<center>METODOLOGÍA DE ANÁLISIS


###**BÚSQUEDA DE BASES DE DATOS REVISADAS PARA PROBAR LA HIPÓTESIS**
https://www.kaggle.com/imnikhilanand/heart-attack-prediction

https://www.kaggle.com/mathchi/diabetes-data-set

https://www.kaggle.com/kashnitsky/a2-demo-analyzing-cardiovascular-data

https://www.kaggle.com/kashnitsky/mlcourse?select=mlbootcamp5_train.csv


###**ELECCIÓN DE LA BASE DE DATOS**
De las bases de datos revisadas, se eligió la que se muestra a continuación para realizar el análisis. Puesto que contiene datos suficientes sobre los factores de riesgo que destacan en el desarrollo de ECV.

https://www.kaggle.com/kashnitsky/mlcourse?select=mlbootcamp5_train.csv



In [1]:
#Iniciamos realizando la carga de paqueterias que seran de utilidad.
import pandas as pd
import requests
import numpy as np
import math

## POSTWORK 3

Una vez elegida la base de datos útil para realizar los análisis, se procedio con la carga de esta.

In [2]:
#carga de datos
df = pd.read_csv('https://raw.githubusercontent.com/angyf/Entrega_Final_Procesamiento_DanielAlvarez_AngelicaFuentes_IvonneRamirez/main/mlbootcamp5_train.csv', sep=',')
df


id    age  gender  height  weight  ...  gluc  smoke  alco  active  cardio
0          0  18393       2     168    62.0  ...     1      0     0       1       0
1          1  20228       1     156    85.0  ...     1      0     0       1       1
2          2  18857       1     165    64.0  ...     1      0     0       0       1
3          3  17623       2     169    82.0  ...     1      0     0       1       1
4          4  17474       1     156    56.0  ...     1      0     0       0       0
...      ...    ...     ...     ...     ...  ...   ...    ...   ...     ...     ...
69995  99993  19240       2     168    76.0  ...     1      1     0       1       0
69996  99995  22601       1     158   126.0  ...     2      0     0       1       1
69997  99996  19066       2     183   105.0  ...     1      0     1       0       1
69998  99998  22431       1     163    72.0  ...     2      0     0       0       1
69999  99999  20540       1     170    72.0  ...     1      0     0       1       0

[70000 rows x 13 columns]

## POSTWORK 4
### Exploración general de la base de datos para su limpieza.

1. ¿El conjunto de datos que tengo realmente me sirve para responder algunas de las preguntas que me planteé?

    Sí, debido a que contiene información para realizar un análisis útil para responder las preguntas iniciales.


2. ¿Qué tamaño tiene mi conjunto de datos? ¿Serán datos suficientes?

  El dataframe, cuenta con 70,000 registros y 13 campos.
  
  Consideramos que el número de registros es suficientepara hacer un análisis. 

In [3]:
#Veamos las dimensiones de nuestro dataframe
df.shape

(70000, 13)

3. ¿Qué columnas tengo y qué información tengo en cada una de esas columnas? 

* El primer campo (id) consiste solamente en un identificador para los 
  registros.

* El campo (age) contiene información sobre la edad de la persona en días.

* El campo (height) contiene información sobre el peso de las personas con unidades de medida en kilogramos.

* El campo (gender) indica el sexo de la persona, donde 1 es mujer y 2 es hombre

* El campo (weight) contiene información sobre la altura de las personas con unidad de medida en centimetros.

* El campo (ap_hi) representa el nivel de presión sistólica medida en milímetros de 
  mercurio.

* El campo (ap_lo) representa el nivel de presión diastólica medida en milímetros de 
  mercurio.

* El campo (cholesterol) indica el nivel de colesterol 1: normal, 2: por arriba de lo normal y 3: lejos de lo normal

* El campo (gluc) indica el nivel de glucosa 1: normal, 2: por arriba de lo normal y 3: lejos de lo normal

* El campo (smoke) indica si la persona fuma o no fuma.

* El campo (alco) indica si la persona consume o no alcohol.
  
* El campo (active) indica si la persona realiza o no actividad 
  física.

* El campo (cardio) indica si la persona 
  tiene o no una enfermedad cardiovascular.


4.  Los nombres que tienen mis **columnas**, ¿son el nombre más apropiado?

  No todas,ya que en las columnas ap_hi y ap_lo los nombres no aportan 
  información por si solas, así mismo consideramos que sería mejor nombrar las columnas en español.

In [4]:
#Consultamos el nombre de las columnas de la tabla
df.columns


Index(['id', 'age', 'gender', 'height', 'weight', 'ap_hi', 'ap_lo',
       'cholesterol', 'gluc', 'smoke', 'alco', 'active', 'cardio'],
      dtype='object')

5. ¿Qué tipos de datos tengo en cada columna? 

  Todos los tipos de datos excepto weight son enteros y seran tratados como 
  tal, sin embargo algunos de ellos representan datos categoricos como lo
  son gender, cholesterol:cardio.

In [5]:
#Tipos de datos que contiene la tabla
df.dtypes

id               int64
age              int64
gender           int64
height           int64
weight         float64
ap_hi            int64
ap_lo            int64
cholesterol      int64
gluc             int64
smoke            int64
alco             int64
active           int64
cardio           int64
dtype: object

6. ¿Parecen ser el tipo correcto de datos? ¿O es un tipo de datos "incorrecto"?

  Consideramos que los tipos de datos son correctos,sin embargo notamos que la 
  variable weight es considerada como float a pesar de contener puros números enteros, sin embargo esto no afectara el análisis pues son datos númericos.

7. Si selecciono algunas filas al azar y las observo, ¿estoy obteniendo los datos que debería? ¿o hay datos que parecen estar "sucios" o "incorrectos"?

  Al parecer los registros no presentan datos sucios, además no hay datos faltantes en nuestro data set

In [6]:
#Información general de los registros
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 70000 entries, 0 to 69999
Data columns (total 13 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   id           70000 non-null  int64  
 1   age          70000 non-null  int64  
 2   gender       70000 non-null  int64  
 3   height       70000 non-null  int64  
 4   weight       70000 non-null  float64
 5   ap_hi        70000 non-null  int64  
 6   ap_lo        70000 non-null  int64  
 7   cholesterol  70000 non-null  int64  
 8   gluc         70000 non-null  int64  
 9   smoke        70000 non-null  int64  
 10  alco         70000 non-null  int64  
 11  active       70000 non-null  int64  
 12  cardio       70000 non-null  int64  
dtypes: float64(1), int64(12)
memory usage: 6.9 MB


In [7]:
df.isna().sum()

id             0
age            0
gender         0
height         0
weight         0
ap_hi          0
ap_lo          0
cholesterol    0
gluc           0
smoke          0
alco           0
active         0
cardio         0
dtype: int64

In [8]:
#visualizaciones de nuestros datos.
df.head(10)

id    age  gender  height  weight  ...  gluc  smoke  alco  active  cardio
0   0  18393       2     168    62.0  ...     1      0     0       1       0
1   1  20228       1     156    85.0  ...     1      0     0       1       1
2   2  18857       1     165    64.0  ...     1      0     0       0       1
3   3  17623       2     169    82.0  ...     1      0     0       1       1
4   4  17474       1     156    56.0  ...     1      0     0       0       0
5   8  21914       1     151    67.0  ...     2      0     0       0       0
6   9  22113       1     157    93.0  ...     1      0     0       1       0
7  12  22584       2     178    95.0  ...     3      0     0       1       1
8  13  17668       1     158    71.0  ...     1      0     0       1       0
9  14  19834       1     164    68.0  ...     1      0     0       0       0

[10 rows x 13 columns]

In [9]:
df.tail(10)

id    age  gender  height  weight  ...  gluc  smoke  alco  active  cardio
69990  99986  15094       1     168    72.0  ...     1      0     0       1       1
69991  99988  20609       1     159    72.0  ...     2      0     0       1       0
69992  99990  18792       1     161    56.0  ...     1      0     0       1       1
69993  99991  19699       1     172    70.0  ...     1      0     0       1       1
69994  99992  21074       1     165    80.0  ...     1      0     0       1       1
69995  99993  19240       2     168    76.0  ...     1      1     0       1       0
69996  99995  22601       1     158   126.0  ...     2      0     0       1       1
69997  99996  19066       2     183   105.0  ...     1      0     1       0       1
69998  99998  22431       1     163    72.0  ...     2      0     0       0       1
69999  99999  20540       1     170    72.0  ...     1      0     0       1       0

[10 rows x 13 columns]

## POSTWORK 5
### Limpieza de la base.

In [10]:
#eliminamos la columna id pues no es de utilidad
df=df.drop(columns='id')

#convertimos edad a años para mejor manejo del dato
df['age']=df['age']/365

#redondeamos la edad a años cumplidos
import math
df['age']=list(map(math.floor,df['age']))

In [11]:


#renombramos las columnas para identificarlas de mejor manera
nombres={
    'age':'edad',
    'gender':'sexo',
    'height':'altura',
    'weight':'peso',
    'ap_hi':'presion_sistolica',
    'ap_lo':'presion_diastolica',
    'cholesterol':'colesterol',
    'gluc':'glucosa',
    'smoke':'fumador',
    'alco':'alcoholico',
    'active':'actividad_fisica',
    'cardio':'enfermedad_cardiaca'}
#dataframe limpio 
datos_adultos=df.rename(columns=nombres)


In [12]:
#Hacemos uso de agregaciones para obtener información general de nuestra población 
prom_edad=round(datos_adultos['edad'].mean(),2)
prom_altura=round(datos_adultos['altura'].mean(),2)
prom_peso=round(datos_adultos['peso'].mean(),2)
prom_sistolica=round(datos_adultos['presion_sistolica'].mean(),2)
prom_diastolica=round(datos_adultos['presion_diastolica'].mean(),2)

print(f'INFORMACIÓN GENERAL DE LA POBLACIÓN \n')
print(f'{("Variables"):20} | {("Promedios"):2}')
print(f'--------------------------------')
print( f'{("Edades"):20} | {prom_edad:2}')
print( f'{("Alturas"):20} | {prom_altura:2}')
print( f'{("Peso"):20} | {prom_peso:2}')
print( f'{("Presion sistolica"):20} | {prom_sistolica:2}')
print( f'{("Presion diastolica"):20} | {prom_diastolica:2}')




INFORMACIÓN GENERAL DE LA POBLACIÓN 

Variables            | Promedios
--------------------------------
Edades               | 52.84
Alturas              | 164.36
Peso                 | 74.21
Presion sistolica    | 128.82
Presion diastolica   | 96.63


#POSTWORK 6

##<center>SUMANDO UNA API CON DATOS DE UNA POBLACIÓN INFANTIL
<div align="justify">

Optamos por la búsqueda de una base complementaria a la que ya teníamos. En este caso encontramos esta API que nos brinda información acerca de datos clínicos de nuestro interés de una población infantil.
Es importante tomar en cuenta a dicha población puesto que las ECV como vemos, no solo afectan a los adultos, sino también a los niños siendo que México ocupa el **primer lugar en obesidad infantil**, y la obesidad es uno de los riesgos más relacionados al desarrollo de ECV.

Este análisis complementario nos permitirá realizar un análisis más completo, aplicable a toda la población afectada.

In [13]:
# Realizamos la carga de los datos de la API que complementaran nuestros análisis.
endpoint='http://data.ny.gov/resource/8itk-gcdy.json?$limit=175000&'
#Esta API no necesita de una api_key por lo que basta con el endpoint
r=requests.get(endpoint)
#Revisamos el estatus de la conexión con la API
r.status_code

200

In [14]:
#Una vez realizada con éxito la conexión obtenemos la consulta en formato Json
response=r.json()
# Para un mejor manejo de los datos se guardan en un Dataframe
datos=pd.DataFrame(response)

In [15]:
#Visualizamos todas las columnas de nuestro dataframe
datos.columns

Index(['survey_year', 'program_category', 'region_served', 'age_group', 'sex',
       'transgender', 'sexual_orientation', 'hispanic_ethnicity', 'race',
       'living_situation', 'household_composition', 'preferred_language',
       'veteran_status', 'employment_status', 'number_of_hours_worked_each',
       'education_status', 'special_education_services', 'mental_illness',
       'intellectual_disability', 'autism_spectrum',
       'other_developmental_disability', 'alcohol_related_disorder',
       'drug_substance_disorder', 'mobility_impairment_disorder',
       'hearing_visual_impairment', 'hyperlipidemia', 'high_blood_pressure',
       'diabetes', 'obesity', 'heart_attack', 'stroke', 'other_cardiac',
       'pulmonary_asthma', 'alzheimer_or_dementia', 'kidney_disease',
       'liver_disease', 'endocrine_condition', 'neurological_condition',
       'traumatic_brain_injury', 'joint_disease', 'cancer',
       'other_chronic_med_condition', 'no_chronic_med_condition',
       'unknow

In [16]:
#Conservamos solo las columnas utiles para el análisis
datos=datos[[ 'age_group', 'sex','alcohol_related_disorder','hyperlipidemia', 'high_blood_pressure','diabetes', 'obesity', 'heart_attack', 'stroke', 'other_cardiac','smokes']]

### Limpieza de la base de datos obtenida de la API

In [17]:
# Veamos si tenemos valores nulos
datos.isna().sum()

age_group                   0
sex                         0
alcohol_related_disorder    0
hyperlipidemia              0
high_blood_pressure         0
diabetes                    0
obesity                     0
heart_attack                0
stroke                      0
other_cardiac               0
smokes                      0
dtype: int64

In [18]:
#No hay datos nulos pero si desconocidos que necesitan ser eliminados del dataframe 
columnas=datos.columns
for columna in columnas:
  print((datos[columna]=='UNKNOWN').sum())

33
194
7420
21253
21253
21253
21253
21253
21253
21253
8413


In [19]:
# Se crea una función para convertir los datos desconocidos a nulos para manejarlos de manera mas sencilla
def repl(x):
  if x=='UNKNOWN':
    return np.nan
  else:
    return x

#aplicamos la función anterior a todas las columnas del dataframe
for columna in columnas:
  datos[columna]=datos[columna].apply(repl)

In [20]:
# De nuevo revisamos cuantos datos nulos tenemos.
datos.isna().sum()

age_group                      33
sex                           194
alcohol_related_disorder     7420
hyperlipidemia              21253
high_blood_pressure         21253
diabetes                    21253
obesity                     21253
heart_attack                21253
stroke                      21253
other_cardiac               21253
smokes                       8413
dtype: int64

In [21]:
# Eliminamos los registros con al menos un dato nulo ya que no podemos sustituirlos por algun otro
datos=datos.dropna(axis=0, how='any')

## POSTWORK 7

Transformación, filtración y ordenamiento de los datos.

In [22]:
#Hacemos la conversión a entero de la columna peso.
datos_adultos['peso']=datos_adultos['peso'].astype(int)

#Agregamos la columna donde calculamos el índice de masa corporal
datos_adultos['imc']=round(datos_adultos['peso']/((datos_adultos['altura']/100)**2),2)

In [23]:
#Creamos nuevos dataset de personas con y sin enfermedad cardiovascular
adultos_enfermos=datos_adultos[datos_adultos['enfermedad_cardiaca']==1]
adultos_sanos=datos_adultos[datos_adultos['enfermedad_cardiaca']==0]

#eliminamos la columna enfermedad_cardiovascular pues ya estan separados
adultos_enfermos=adultos_enfermos.drop(columns='enfermedad_cardiaca')
adultos_sanos=adultos_sanos.drop(columns='enfermedad_cardiaca')

In [24]:
#De estos nuevos datos mantenemos los de los infantes para el análisis
datos_infantes=datos[datos['age_group']=='CHILD']

# Borramos la columna de grupos de edad pues ya no es necesaria
del datos_infantes['age_group']

#Renombramos las columnas 
nombres={
     'sex':'sexo', 'alcohol_related_disorder':'alcoholismo', 'hyperlipidemia':'hiperlipidemia',
       'high_blood_pressure':'presion_alta', 'obesity':'obesidad', 'heart_attack':'ataque_cardiaco', 'stroke':'derrame_cerebrovascular',
       'other_cardiac':'otra_enfermedad_cardiaca', 'smokes':'tabaquismo'
}
datos_infantes=datos_infantes.rename(columns=nombres)

In [25]:
#Cambiamos el tipo de dato del campo sexo (de string a int), para mantener la homogeneidad entre las bases
# 1: NIÑA    2: NIÑO
str_a_int = {
    'FEMALE' : int(1),
    'MALE': int(2),
}

datos_infantes['sexo'] = datos_infantes['sexo'].map(str_a_int)


### AGRUPAMIENTOS PARA COMENZAR A RESPONDER ALGUNAS DE LAS PREGUNTAS PLANTEADAS

In [26]:
#¿Qué % de hombres y mujeres hay?
# 1: MUJER   2: HOMBRE
AT=len(datos_adultos)
print(round((datos_adultos['sexo'].value_counts())/ AT * 100,2))

1    65.04
2    34.96
Name: sexo, dtype: float64


In [27]:
#¿Cuántos adultos enfermos hay por sexo?
#1: MUJER  2:HOMBRE
enf=adultos_enfermos.groupby('sexo').size()
AET=len(adultos_enfermos)
print(enf)
print()
#¿Qué % representa?
print(f'{round(enf/ AET*100,2)}')

sexo
1    22616
2    12363
dtype: int64

sexo
1    64.66
2    35.34
dtype: float64


In [28]:
#¿Cuántos adultos sanos hay por sexo?
# 1: MUJER   2: HOMBRE
AST=len(adultos_sanos)
san=adultos_sanos.groupby('sexo').size()
print(san)
print()
#¿Qué % representa?
print(f'{round(san/ AST*100,2)}')

sexo
1    22914
2    12107
dtype: int64

sexo
1    65.43
2    34.57
dtype: float64


In [29]:
#¿Cuántos adultos sufren de niveles de colesterol alto?
adul_col_al=(adultos_enfermos[adultos_enfermos['colesterol'] == 3]['colesterol']).count()
print(adul_col_al)

#¿Qué % representa?
print(f'{adul_col_al / AT*100} %')

6174
8.82 %


In [30]:
#¿Cuántos adultos sufren de niveles de glucosa alta?
adul_glu_al=(adultos_enfermos['glucosa']== 3).sum()
print(adul_glu_al)
#¿Qué % representa?
print(f'{round(adul_glu_al / AT*100,2)} %')

3316
4.74 %


In [31]:
#¿Cuántos adultos fuman en total?
adul_fum=(datos_adultos['fumador'] == 1).sum()
print(adul_fum)
#¿Qué % representa?
print(f'{round(adul_fum / AT*100)} %')

6169
9 %


In [32]:
#¿Cuántas mujeres y cuántos hombres fuman?
# 1: MUJER   2: HOMBRE
# 0: NO FUMA  1: SI FUMA
datos_adultos.groupby('sexo')['fumador'].value_counts()

sexo  fumador
1     0          44717
      1            813
2     0          19114
      1           5356
Name: fumador, dtype: int64

In [33]:
#¿Cuántas personas alcohólicas hay?
adul_alc=(datos_adultos['alcoholico'] == 1).sum()
print(adul_alc)
#¿Qué % representa?
print(f'{round(adul_alc / AT*100,2)} %')

3764
5.38 %


In [34]:
#¿Cuántas personas alcohólicas hay por sexo?
# 1: MUJER   2: HOMBRE
# 0: NO TOMA    1: SI TOMA
datos_adultos.groupby('sexo')['alcoholico'].value_counts()

sexo  alcoholico
1     0             44369
      1              1161
2     0             21867
      1              2603
Name: alcoholico, dtype: int64

In [35]:
#¿Cuántas personas realizan actividad física?
adul_actf=(datos_adultos['actividad_fisica'] == 1).sum()
print(adul_actf)
#¿Qué % representa?
print(f'{round(adul_actf / AT*100,2)} %')

56261
80.37 %


###<center>AGRUPAMIENTOS DE LA API: POBLACIÓN INFANTIL
<div align="justify">
Realizamos agrupamientos para poder contestar preguntas similares a las hechas en la población adulta.
Esto para tener una idea general de cómo se encuentra esta población, y ver cómo se encuentran afectados por los factores de riesgo asociados al desarrollo de ECV.

In [36]:
#¿Qué % de niños hay?
# 1: NIÑA    2: NIÑO
NT = len(datos_infantes)
print(round((datos_infantes['sexo'].value_counts())/ NT * 100,2))

2    53.37
1    46.63
Name: sexo, dtype: float64


In [37]:
#¿Cuántos infantes son alcohólicos?
niños_alc = (datos_infantes['alcoholismo'] == 'YES').sum()
print(niños_alc)
#¿Qué % representa?
print(f'{round(niños_alc / NT*100,2)} %')

170
0.58 %


In [38]:
#¿Cuántos niños presentan obesidad?
niños_obesos = (datos_infantes['obesidad'] == 'YES').sum()
print(niños_obesos)
#¿Qué % representa?
print(f'{round(niños_obesos / NT*100,2)} %')

1580
5.41 %


In [39]:
#¿Cuántos niños son fumadores?
niños_f =(datos_infantes['tabaquismo'] == 'YES').sum()
print(niños_f)
#¿Qué % representa?
print(f'{round(niños_f / NT*100,2)} %')

589
2.02 %


In [40]:
#¿Cuántos prsentan hiperlipidemia? (grasa en la sangre)
niños_hiper =(datos_infantes['hiperlipidemia'] == 'YES').sum()
print(niños_hiper)
#¿Qué % representa?
print(f'{round(niños_hiper / NT*100,2)} %')

259
0.89 %


In [41]:
#¿Cuántos niños presentan presión alta?
niños_pres =(datos_infantes['presion_alta'] == 'YES').sum()
print(niños_pres)
#¿Qué % representa?
print(f'{round(niños_pres / NT*100,2)} %')

119
0.41 %


In [42]:
#¿Cuántos niños han sufrido un ataque al corazón?
niños_ataque =(datos_infantes['ataque_cardiaco'] == 'YES').sum()
print(niños_ataque)
#¿Qué % representa?
print(f'{round(niños_ataque / NT*100,2)} %')

4
0.01 %


In [43]:
#¿Cuántos niños han sufrido un derrame cerebrovascular?
niños_derrame =(datos_infantes['derrame_cerebrovascular'] == 'YES').sum()
print(niños_derrame)
#¿Qué % representa?
print(f'{round(niños_derrame / NT*100,2)} %')

11
0.04 %


In [44]:
#¿Cuántos niños han sufrido otra enfermedad cadiaca?
niños_otra =(datos_infantes['otra_enfermedad_cardiaca'] == 'YES').sum()
print(niños_otra)
#¿Qué % representa?
print(f'{round(niños_otra / NT*100,2)} %')

177
0.61 %


#POSTWORK 8 

<p align="justify">Con este análisis exploratorio fue posible conocer mas acerca de las características generales de nuestras poblaciones, tanto de adultos como de infantes. Cabe resaltar que decidimos analizar población infantil ya que México ocupa el primer lugar de obsesidad infantil, por lo que es importante tomar en cuenta a este grupo vulnerable, esto para poder plantear estrategias que ayuden a disminuir la incidencia de la enfermedad de manera general.

<p align="justify">Además, se pudieron resolver algunas preguntas planteadas al inicio del análisis sobre el número de personas que presentan alguna enfermedad cardiovascular, el porcentaje de hombres y mujeres, fumadores y no fumadores, entre otras. Si bien estos porcentajes hablan por sí mismos, resulta necesario contar con una forma de mostrarlos graficamente.

<p align="justify">También consideramos que estos datos nos serán de utilidad para corroborar las conclusiones a las que se llegaron anteriormente sobre los factores que influyen de manera significativa al desarrollo de enfermedades cardiovasculares.

En cuanto a los planes a futuro se tiene:

<div align="justify">

*   Presentar graficamente hallazgos realizados en este análisis para una mejor visualización y descripción de nuestra población en relación a cada factor de riesgo.
*   Validar la aplicación del modelo predictivo previamente planteado.
* Realizar un anáisis estadístico más amplio con el fin de establecer un modelo predictivo ajustado para la población infantil.
* Dependiendo de los resultados, sugerir replantear estrategias específicas para reducir el índice de morbilidad y mortalidad.
*Detectar a pacientes en grupos de alto riesgo, con el fin de predecir un diagnóstico temprano del desarrollo de la enfermedad.
</p>







#<center>REFERENCIAS


<div align="justify">
	1. Instituto Nacional de Estadística y Geografía (INEGI). Comunicado de prensa NÚM. 61/21 “Características de las defunciones registradas en México durante enero a agosto de 2020” https://www.inegi.org.mx/contenidos/saladeprensa/boletines/2021/EstSociodemo/DefuncionesRegistradas2020_Pnles.pdf 
<div align="justify">
	2. Secretaría de Salud. “Programa de Acción Específico. Prevención y Control de la Obesidad y Riesgo Cardiovascular. Programa Sectorial De Salud 2013-2018”. http://www.cenaprece.salud.gob.mx/descargas/pdf/PAE_PrevencionControlObesidadRiesgoCardiovascular2013_2018.pdf 
<div align="justify">
	3. Secretaría de Salud (2001). “Programa de Acción Enfermedades Cardiovasculares e Hipertensión Arterial” http://www.salud.gob.mx/unidades/cdi/documentos/enf_cardiovasculares.pdf 
<div align="justify">
	4. Dávila CCA. Tendencia e impacto de la mortalidad por enfermedades cardiovasculares en México, 1990-2015. Revista Cubana de Salud Pública. 2019;45(4):1-18. https://www.scielosp.org/article/rcsp/2019.v45n4/e1081/# 
<div align="justify">
	5. Encuesta Nacional de Salud y Nutrición (ENSANUT) 2018 Presentación de resultados https://ensanut.insp.mx/encuestas/ensanut2018/doctos/informes/ensanut_2018_presentacion_resultados.pdf 
  <div align="justify"> 
	6. Sánchez Arias, A., Bobadilla Serrano, M., Dimas Altamirano, B., Gómez Ortega, M, et al. (2016). “Enfermedad cardiovascular: primera causa de morbilidad en un hospital de tercer nivel”. Revista Mexicana de Cardiología. https://www.medigraphic.com/pdfs/cardio/h-2016/hs163a.pdf
  <div align="justify">
	7. World Heart Federation. “FACTSHEET: Enfermedades Cardiovasculares en México”. https://www.world-heart-federation.org/wp content/uploads/2017/05/Cardiovascular_diseases_in_Mexico__Spanish_.pdf
  <div align="justify">
	8. Instituto Nacional de Estadística y Geografía (INEGI). Comunicado de prensa NÚM. 525/18 “Características de las defunciones registradas en México durante 2017” https://www.inegi.org.mx/contenidos/saladeprensa/boletines/2018/EstSociodemo/DEFUNCIONES2017.pdf 
  <div align="justify">
	9. Instituto Nacional de Estadística y Geografía (INEGI). Comunicado de prensa NÚM. 538/19 “Características de las defunciones registradas en México durante 2018” https://www.inegi.org.mx/contenidos/saladeprensa/boletines/2019/EstSociodemo/DefuncionesRegistradas2019.pdf 
  <div align="justify">
	10. Instituto Nacional de Estadística y Geografía (INEGI). Nota técnica. “Estadística de defunciones registradas 2019” https://www.inegi.org.mx/contenidos/programas/mortalidad/doc/defunciones_registradas_2019_nota_tecnica.pdf 
  <div align="justify">
	11. Lau WB, Ohashi K, Wang Y, Ogawa H, Murohara T, Ma XL, et al. Role of adipokines in cardiovascular disease. Circ J. 2017;81(7): 920–8. https://doi.org/10.1253/circj.CJ-17-0458.
  <div align="justify">
	12. Global Obesity Health Collaborators. Health effects of overweight and obesity in 195 countries over 25 years. N Engl J Med. 2017;377(1):13–27. https://doi.org/10.1056/NEJMc1710026.
  <div align="justify">
	13. https://www.insp.mx/avisos/4884-la-obesidad-mexico.html “La obesidad en México. Estado de la política pública y recomendaciones para su prevención y control”.
  <div align="justify">
	14. https://www.who.int/es/news-room/fact-sheets/detail/obesity-and-overweight
  <div align="justify">
	15. Lavie CJ, McAuley PA, Church TS, Milani RV, Blair SN. Obesity and cardiovascular diseases: implications regarding fitness, fatness, and severity in the obesity paradox. J Am Coll Cardiol. 2014;63(14):1345–54. https://doi.org/10.1016/j.jacc.2014.01.022.
  <div align="justify">
	16. Ortega FB, Lavie CJ, Blair SN. Obesity and cardiovascular disease. Circ Res. 2016;118(11):1752–70. https://doi.org/10.1161/ CIRCRESAHA.115.306883.
  <div align="justify">
	17. Caleyachetty R, Thomas GN, Toulis KA, Mohammed N, Gokhale KM, Balachandran K, et al. Metabolically healthy obese and inci- dent cardiovascular disease events among 3.5 million men and women. J Am Coll Cardiol. 2017;70(12):1429–37. https://doi.org/ 10.1016/j.jacc.2017.07.763.
  <div align="justify">
	18. Ortega FB, Cadenas-Sanchez C, Migueles JH, Labayen I, Ruiz JR, Sui X, et al. Role of physical activity and fitness in the char- acterization and prognosis of the metabolically healthy obesity phenotype: a systematic review and meta-analysis. Prog Cardiovasc Dis. 2018;61(2):190–205. https://doi.org/10.1016/j. pcad.2018.07.008.
  <div align="justify">
	19. Zheng R, Liu C, Wang C, Zhou B, Liu Y, Pan F, et al. Natural course of metabolically healthy overweight/obese subjects and the impact of weight change. Nutrients. 2016;8(7):430. https://doi.org/ 10.3390/nu8070430.
  <div align="justify">
	20. De Marco VG, Aroor AR, Sowers JR. The pathophysiology of hy- pertension in patients with obesity. Nat Rev Endocrinol. 2014;10(6):364–76. https://doi.org/10.1038/nrendo.2014.44.
  <div align="justify">
	21. Whelton PK, Carey RM, Aronow WS, Casey DE Jr, Collins KJ, Dennison Himmelfarb C, et al. 2017 ACC/AHA/AAPA/ABC/ ACPM/AGS/APhA/ ASH/ASPC/NMA/PCNA guideline for the prevention, detection, evaluation, and management of high blood pressure in adults a report of the American College of Cardiology/ American Heart Association task force on clinical pr. Hypertension. 2018;71(6):e13–e115. https://doi.org/10.1161/HYP.
  <div align="justify">
	22. McGill HC, McMahan CA, Herderick EE, et al. Obesity accelerates the progression of coronary atherosclerosis in young men. Circulation. 2002;105(23):2712–8. https://doi.org/10.1161/01. CIR.0000018121.67607.CE.
  <div align="justify">
	23.  Jahangir E, De Schutter A, Lavie CJ. Low weight and overweightness in older adults: risk and clinical management. Prog Cardiovasc Dis. 2014;57(2):127–33. https://doi.org/10. 1016/j.pcad.2014.01.001.
  <div align="justify">
	24. Lavie CJ, Milani RV, Artham SM, Patel DA, Ventura HO. The obesity paradox, weight loss, and coronary disease. Am J Med. 2009;122(12):1106–14. https://doi.org/10.1016/j.amjmed.2009. 06.006.
  <div align="justify">
	25. Kenchaiah S, Chesebro J. The epidemiologic association between obesity and heart failure. Am Coll Cardiol Ext Learn. 2017;49(8): 4–6.
  <div align="justify">
	26. Swift DL, McGee JE, Earnest CP, Carlisle E, Nygard M, Johannsen NM. The effects of exercise and physical activity on weight loss and maintenance. Prog Cardiovasc Dis. 2018;61(2):206–13. https:// doi.org/10.1016/j.pcad.2018.07.014.
  <div align="justify">
	27. Piercy KL, Troiano RP, Ballard RM, Carlson SA, Fulton JE, Galuska DA, et al. The physical activity guidelines for Americans. JAMA. 2018;320(19):2020–8. https://doi.org/10. 1001/jama.2018.14854.
  <div align="justify">
	28. Organización Mundial de la Salud. Enfermedades cardiovasculares. 2017. https://www.who.int/es/news-room/fact-sheets/detail/cardiovascular-diseases-(cvds).
  <div align="justify">
	29. Centers for Disease Control and Prevention (US); National Center for Chronic Disease Prevention and Health Promotion (US); Office on Smoking and Health (US). How Tobacco Smoke Causes Disease: The Biology and Behavioral Basis for Smoking-Attributable Disease: A Report of the Surgeon General. Atlanta (GA): Centers for Disease Control and Prevention (US); 2010. 6, Cardiovascular Diseases. https://www.ncbi.nlm.nih.gov/books/NBK53012/#_ch6_s101_
  <div align="justify">
	30. Secretaría de Salud. Fumar también enferma tu corazón. https://www.gob.mx/salud/articulos/fumar-tambien-enferma-tu-corazon?idiom=es
  <div align="justify">
	31. Shamah-Levy T, Vielma-Orozco E, Heredia-Hernández O, Romero-Martínez M, Mojica-Cuevas J, Cuevas-Nasu L, Santaella-Castell JA, Rivera-Dommarco J. Encuesta Nacional de Salud y Nutrición 2018-19: Resultados Nacionales. Cuernavaca, México: Instituto Nacional de Salud Pública, 2020.  https://ensanut.insp.mx/encuestas/ensanut2018/doctos/informes/ensanut_2018_informe_final.pdf
  <div align="justify">
	32. Centers for Disease Control and Prevention (US); National Center for Chronic Disease Prevention and Health Promotion (US); Office on Smoking and Health (US). How Tobacco Smoke Causes Disease: The Biology and Behavioral Basis for Smoking-Attributable Disease: A Report of the Surgeon General. Atlanta (GA): Centers for Disease Control and Prevention (US); 2010. 6, Cardiovascular Diseases.  https://www.ncbi.nlm.nih.gov/books/NBK53012/#_ch6_s101_
  <div align="justify">
	33. Organización Panamericana de la Salud. Control del tabaco. https://www.paho.org/es/temas/ prevencion-control-tabaco 
  <div align="justify">
	34. Organización Mundial de la Salud.El consumo nocivo de alcohol mata a más de 3 millones de personas al año, en su mayoría hombres.2018. https://www.who.int/es/newsitem21-09-2018-harmful-use-of-alcohol-kills-more-than-3-million-people-each-year--most-of-them-men
  <div align="justify">
	35. Anderson P, Gual A, Colon J. Alcohol y atención primaria de la salud: informaciones clínicas básicas para la identificación y el manejo de riesgos y problemas. 2008. Washington, D.C.: OPS. https://www.who.int/substance_abuse/publications/alcohol_atencion_primaria.pdf
  <div align="justify">
	36. Organización Mundial de la Salud. Estrategia mundial sobre régimen alimentaria, actividad física y salud. Dieta.  https://www.who.int/dietphysicalactivity/diet/es/
  <div align="justify">
	37. Organización Mundial de la Salud. Reducir el consumo de sal. 2020. https://www.who.int/es/news- room/fact-sheets/detail/salt-reduction
  <div align="justify">
	38. Organización Panamericana de la Salud.Nutrición. https://www.paho.org/es/temas/nutricion
  <div align="justify">
	39. UNICEF, México. UNICEF: El etiquetado frontal de alimentos y bebidas aprobado en México, “de los mejores del mundo”.2020. https://www.unicef.org/mexico/comunicados-prensa/unicef-el-etiquetado-frontal-de-alimentos-y-bebidas-aprobado-en-méxico-de-los


